rosbag2_tests/test/rosbag2_tests/test_rosbag2_cpp_get_service_info.cpp

```diff
@@ -222,8 +228,6 @@ TEST_P(Rosbag2CPPGetServiceInfoTest, get_service_info_for_bag_with_services_only
   EXPECT_TRUE(ret) << "Failed to capture " << expected_messages << " expected messages in time";
 
   recorder->stop();
-  stop_spinning();
-  cleanup_process_handle.cancel();
```

Comment:
Suggestion to return deleted explict call to `stop_spinning()` and `cleanup_process_handle.cancel();`

Reply:
I ended up doing something slightly different, because I think the pattern of calling `stop_spinning` and `cancel` on the scope exit isn't great.  Just for the few instances you pointed out here, I ended up making scopes.  That way, when we leave the scope, `cleanup_process_handle` will always run.  So effectively [e63f8c7](https://github.com/ros2/rosbag2/pull/1796/commits/e63f8c7b27f139e011f4452f9a9fbd795be0e0a2) does what you ask here, just in a slight different way.  Let me know what you think.

Reply:
Yeah, that is even better I think.